docs/architecture/adr-005-consensus-height-events.md

```diff
@@ -0,0 +1,83 @@
+# ADR 005: UpdateClient Events - ClientState Consensus Heights
+
+## Changelog
+* 25/04/2022: initial draft
+
+## Status
+
+Accepted 
+
+## Context
+
+The `ibc-go` implementation leverages the [Cosmos-SDK's EventManager](https://github.com/cosmos/cosmos-sdk/blob/main/docs/core/events.md#EventManager) to provide subscribers a method of reacting to application specific events.
+IBC relayers depend on the [`consensus_height`](https://github.com/cosmos/ibc-go/blob/main/modules/core/02-client/keeper/events.go#L33) attribute emitted as part of `UpdateClient` events in order to run misbehaviour detection by cross-checking the details of the *Header* emitted at a given consensus height against that of the *Header* from the originating chain.
```

Comment:
Maybe also replace this link to main https://github.com/cosmos/ibc-go/blob/main/modules/core/02-client/keeper/events.go#L33 with a tag.